PhysicsTools/PatAlgos/python/slimming/applySubstructure_cfi.py

```diff
@@ -0,0 +1,158 @@
+                     btagInfos = ['caTopTagInfosPAT'],
+                     genJetCollection = cms.InputTag('slimmedGenJetsAK8')
+                     )
+    process.patJetsAK8.userData.userFloats.src = [] # start with empty list of user floats
+    process.selectedPatJetsAK8.cut = cms.string("pt > 50")
+
+
+
+    ## AK8 groomed masses
+    from RecoJets.Configuration.RecoPFJets_cff import ak8PFJetsCHSPruned, ak8PFJetsCHSSoftDrop, ak8PFJetsCHSFiltered, ak8PFJetsCHSTrimmed 
+    process.ak8PFJetsCHSPruned   = ak8PFJetsCHSPruned.clone()
+    process.ak8PFJetsCHSSoftDrop = ak8PFJetsCHSSoftDrop.clone()
+    process.ak8PFJetsCHSTrimmed  = ak8PFJetsCHSTrimmed.clone()
+    process.ak8PFJetsCHSFiltered = ak8PFJetsCHSFiltered.clone()
+    process.load("RecoJets.JetProducers.ak8PFJetsCHS_groomingValueMaps_cfi")
+    process.patJetsAK8.userData.userFloats.src += ['ak8PFJetsCHSPrunedMass','ak8PFJetsCHSSoftDropMass','ak8PFJetsCHSTrimmedMass','ak8PFJetsCHSFilteredMass']
+
+    # Add AK8 top tagging variables
+    process.patJetsAK8.tagInfoSources = cms.VInputTag(cms.InputTag("caTopTagInfosPAT"))
+    process.patJetsAK8.addTagInfos = cms.bool(True)
+
+
+
+    # add Njetiness
+    process.load('RecoJets.JetProducers.nJettinessAdder_cfi')
+    process.NjettinessAK8 = process.Njettiness.clone()
+    process.NjettinessAK8.src = cms.InputTag("ak8PFJetsCHS")
+    process.NjettinessAK8.cone = cms.double(0.8)
+    process.patJetsAK8.userData.userFloats.src += ['NjettinessAK8:tau1','NjettinessAK8:tau2','NjettinessAK8:tau3']
+
+
+    ## PATify pruned fat jets
+    addJetCollection(
+        process,
+        labelName = 'AK8PFCHSSoftDrop',
+        jetSource = cms.InputTag('ak8PFJetsCHSSoftDrop'),
+        btagDiscriminators = ['None'],
+        jetCorrections = ('AK8PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        getJetMCFlavour = False # jet flavor disabled
+    )
+    getattr(process,'patJetPartonMatchAK8PFCHSSoftDrop').matched = cms.InputTag('prunedGenParticles')
+    ## PATify soft drop subjets
+    addJetCollection(
+        process,
+        labelName = 'AK8PFCHSSoftDropSubjets',
+        jetSource = cms.InputTag('ak8PFJetsCHSSoftDrop','SubJets'),
+        algo = 'ak',  # needed for subjet flavor clustering
+        rParam = 0.8, # needed for subjet flavor clustering
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        jetCorrections = ('AK4PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        explicitJTA = True,  # needed for subjet b tagging
+        svClustering = True, # needed for subjet b tagging
+        genJetCollection = cms.InputTag('slimmedGenJets'), 
+        fatJets=cms.InputTag('ak8PFJetsCHS'),             # needed for subjet flavor clustering
+        groomedFatJets=cms.InputTag('ak8PFJetsCHSSoftDrop') # needed for subjet flavor clustering
+    )
+    process.selectedPatJetsAK8PFCHSSoftDrop.cut = cms.string("pt > 200")
```

Comment:
@arizzi The same producer makes groomed fat jets and their subjets and a pt cut that is applied to fat jets at that level is 50 GeV. This also means there won't be any subjets produced for fat jets below 50 GeV. However, for MiniAOD we want a tighter pt cut which means that additional subjets need to be dropped.